umi_tools/whitelist.py

```diff
@@ -42,14 +42,39 @@
 provide the following options: --allow-threshold-error
 --plot-prefix=[PLOT_PREFIX]
 
+Finally, in some dataset there may be a risk that CBs above the
+selected threshold are actually errors from another CB. We can detect
+potential instances of this by looking for CBs within one error
+(substition, insertion or deletion) of another CB with higher
+counts. One can then either take a conservate approach (remove CB with
+lower counts), or a more relaxed approach (correct CB with lower
+counts to CB with higher counts).  Of course, the risk with the
+relaxed approach is that this may erroneously merge two truly
+different CBs together and create an in-silico "doublet". The end of
+the log file (--log) will detail the number of reads from CBs above
+the threshold which may be errors. In most cases, we expect the number
+of reads to be a very small fraction of the total reads and therefore
+recommend taking the conservative approach. See
+https://cgatoxford.wordpress.com/2017/05/23/estimating-the-number-of-true-cell-barcodes-in-single-cell-rna-seq-part-2/
+for an analysis of errors in barcodes above the knee threshold.
+
+Optional detection of putative error CBs above the threshold can be
+switched on with the error detection option:
+--ed-above-threshold
```

Comment:
Why two different options? Why not `--ed-above-thershold=[discard|correct]` with a default setting of False/None?

Reply:
Yeah, that would be better. One other issue is what error threshold to allow for the error detection above the threshold. Right now, it re-uses the `--error-correct-threshold` which is still described as "Hamming distance for correction of barcodes to whitelist barcodes", e.g same error threshold correcting 1) below knee to above knee and 2) between the above knee barcodes. I assume this is the behaviour that users will desire but maybe there will be an edge case where different thresholds are desirable. For now, I'll just update the description for `--error-correct-threshold` to make this clearer.